packages/component/src/Attachment/Text/TextContent.tsx

```diff
@@ -1,30 +1,56 @@
-import React, { type FC, memo } from 'react';
+import React, { memo, useMemo } from 'react';
+import classNames from 'classnames';
+import { hooks } from 'botframework-webchat-api';
+import { type WebChatActivity } from 'botframework-webchat-core';
 
+import isAIGeneratedActivity from './private/isAIGeneratedActivity';
 import MarkdownTextContent from './private/MarkdownTextContent';
 import PlainTextContent from './private/PlainTextContent';
+import CustomPropertyNames from '../../Styles/CustomPropertyNames';
+import useStyleToEmotionObject from '../../hooks/internal/useStyleToEmotionObject';
 import useRenderMarkdownAsHTML from '../../hooks/useRenderMarkdownAsHTML';
 
-import { type WebChatActivity } from 'botframework-webchat-core';
+const { useLocalizer } = hooks;
 
 type Props = Readonly<{
   activity: WebChatActivity;
   contentType?: string;
   text: string;
 }>;
 
-const TextContent: FC<Props> = memo(({ activity, contentType = 'text/plain', text }: Props) => {
+const generatedBadgeStyle = {
+  '&.webchat__text-content__generated-badge': {
+    fontSize: `var(${CustomPropertyNames.FontSizeSmall})`,
+    color: `var(${CustomPropertyNames.ColorSubtle})`
+  }
+};
+
+const TextContent = memo(({ activity, contentType = 'text/plain', text }: Props) => {
   const supportMarkdown = !!useRenderMarkdownAsHTML('message activity');
+  const localize = useLocalizer();
+  const generatedBadgeClassName = useStyleToEmotionObject()(generatedBadgeStyle) + '';
+
+  const generatedBadge = useMemo(
+    () =>
+      isAIGeneratedActivity(activity) && (
+        <div className={classNames('webchat__text-content__generated-badge', generatedBadgeClassName)}>
+          {localize('ACTIVITY_CONTENT_CAUTION')}
+        </div>
+      ),
+    [activity, generatedBadgeClassName, localize]
+  );
 
   return text ? (
     contentType === 'text/markdown' && supportMarkdown ? (
-      <MarkdownTextContent activity={activity} markdown={text} />
+      <MarkdownTextContent activity={activity} markdown={text}>
+        {generatedBadge}
+      </MarkdownTextContent>
     ) : (
-      <PlainTextContent text={text} />
+      <PlainTextContent text={text}>{generatedBadge}</PlainTextContent>
```

Comment:
Should we `<Fragment>` and put the `generatedBadge` below it?

Reply:
We could but we need the same classes, so paddings are preserved. I was thinking it'd be better to reuse classes in the scope of the component.